scripts/missions/amk/06_An_Errand_The_Professors_Price.lua

```diff
@@ -0,0 +1,152 @@
+    end
+    if player:hasKeyItem(xi.ki.ORB_OF_COINS) then
+        player:delKeyItem(xi.ki.ORB_OF_COINS)
+        -- TODO: Remove magic damage immunity
+    end
+
+    if removedKIs == 3 then
+        numToSpawn = 10
+    elseif removedKIs == 4 then
+        numToSpawn = 5
+    end
+
+    xi.confrontation.start(player, npc, cardianIds, function(playerArg)
+        npcUtil.giveKeyItem(playerArg, xi.keyItem.RIPE_STARFRUIT)
+        npcUtil.giveKeyItem(playerArg, xi.keyItem.PEACH_CORAL_KEY)
+    end)
+end
+
+mission.sections =
+{
+    -- Go get the Starfruit
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId and not player:hasKeyItem(xi.ki.RIPE_STARFRUIT)
+        end,
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Shantotto'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(508)
+                end,
+            },
+        },
+
+        [xi.zone.OUTER_HORUTOTO_RUINS] =
+        {
+            ['qm1'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(100)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [100] = function(player, csid, option, npc)
+                    if option == 1 then
+                        beginCardianFight(player, npc)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Got the Starfruit
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId and player:hasKeyItem(xi.ki.RIPE_STARFRUIT) and not player:needToZone()
+        end,
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Shantotto'] =
+            {
+                onTrigger = function(player, npc)
+                    local hasGil = player:getGil() >= 5000 and 0 or 1
+                    return mission:progressEvent(507, 0, player:getGil(), hasGil)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [507] = function(player, csid, option, npc)
+                    if option == 0 then -- Dont Pay
+                        player:needToZone(true)
+                    elseif option == 1 then -- Pay
+                        if mission:complete(player) then
+                            player:delGil(5000)
+                            player:delKeyItem(xi.ki.RIPE_STARFRUIT)
+                        end
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.OUTER_HORUTOTO_RUINS] =
+        {
+            ['qm1'] =
+            {
+                -- TODO: Reminder about the orbs
+                onTrigger = function(player, npc)
+                    return mission:messageSpecial(ID.text.CANNOT_ENTER_BATTLEFIELD, xi.ki.RIPE_STARFRUIT)
```

Comment:
If this is a battlefield qm, this message might not be overriding, and allow for moving back and forth between BCNM entry and that message (messageText, and I think messageSpecial are both much lower than say, event priority)

Reply:
In that case, what should I do here, play a regular messageSpecial and return something else?

Reply:
If it needs to be completely blocking for people on that mission, I'd suggest manually setting the priority.  Something like:
`return mission:messageSpecial(ID.text.CANNOT_ENTER_BATTLEFIELD, xi.ki.RIPE_STARFRUIT):setPriority(1000)`

Here's some of the standard values, so event would take precedence:
```
Action.Priority = {
    Ignore = 1,
    Default = 10,
    ReplaceDefault = 20,
    Message = 50,
    Event = 100,
    Progress = 1000,
}
```

Reply:
Didn't know I could do this, thanks!